apollo-router/src/router_factory.rs

```diff
@@ -237,6 +241,14 @@ impl YamlRouterFactory {
                 extra_plugins,
             )
             .await?;
+
+        // Don't let the router start in experimental_query_planner_mode and
+        // unimplemented Rust QP features.
+        can_use_with_experimental_query_planner(
+            configuration.clone(),
+            supergraph_creator.schema(),
+        )?;
+
```

Comment:
I put this in the YamlRouterFactory, it can perhaps also go right on the QueryAnalysisLayer impl. But that seems to be carried around everywhere, so I opted in for this. Let me know if there is a better spot!